src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -204,11 +219,59 @@ public static Bundle update (Request req, Response res) throws IOException {
         return Persistence.bundles.updateFromJSONRequest(req);
     }
 
-    public static Object getBundle (Request req, Response res) {
-        return Persistence.bundles.findByIdFromRequestIfPermitted(req);
+    public static Bundle getBundle (Request req, Response res) {
+        Bundle bundle = Persistence.bundles.findByIdFromRequestIfPermitted(req);
+
+        // Old bundles did not have these set.
+        try {
+            setBundleServiceDates(bundle);
+        } catch (Exception e) {
+            throw AnalysisServerException.unknown(e);
+        }
+
+        return bundle;
+    }
+
+    public static Collection<Bundle> getBundles (Request req, Response res) {
+        return Persistence.bundles.findPermittedForQuery(req);
+    }
+
+    /**
+     * Old bundles were created without computing the service start and end dates. This finds them and stores them.
+     * @param bundle
+     */
+    public static Bundle setBundleServiceDates (Bundle bundle) throws Exception {
+        if (bundle.status != Bundle.Status.DONE || (bundle.serviceStart != null && bundle.serviceEnd != null)) return bundle;
+
+        // Old bundles were created with computing the service start and end dates
+        bundle.serviceStart = LocalDate.MAX;
+        bundle.serviceEnd = LocalDate.MIN;
+
+        for (Bundle.FeedSummary summary : bundle.feeds) {
+            // Compute the feed start and end dates
+            if (summary.serviceStart == null || summary.serviceEnd == null) {
+                FeedSource fs = ApiMain.getFeedSource(Bundle.bundleScopeFeedId(summary.feedId, bundle._id));
+                // Set service start and end from the dates of service
+                List<LocalDate> datesOfService = fs.feed.getDatesOfService();
+                datesOfService.sort(Comparator.naturalOrder());
+                summary.serviceStart = datesOfService.get(0);
+                summary.serviceEnd = datesOfService.get(datesOfService.size() - 1);
+            }
+
+            if (summary.serviceStart.isBefore(bundle.serviceStart)) {
+                bundle.serviceStart = summary.serviceStart;
+            }
+            if (summary.serviceEnd.isAfter(bundle.serviceEnd)) {
+                bundle.serviceEnd = summary.serviceEnd;
+            }
+        }
+
+        // Automated change that could occur on a `get`, so don't update the nonce
+        return Persistence.bundles.modifiyWithoutUpdatingLock(bundle);
     }
 
     public static void register () {
+        get("/api/bundle", BundleController::getBundles, JsonUtil.objectMapper::writeValueAsString);
```

Comment:
This PR seems to be adding two new API endpoints to fetch all bundles and all projects. Are these changes needed to allow recording the valid date ranges for bundles? Can you add something about this to the PR description?

Reply:
I'll add to the description